CMakeLists.txt

```diff
@@ -85,6 +85,10 @@ mongo_setting(ENABLE_ZSTD "Enable zstd compression support"
 mongo_setting(ENABLE_ZLIB "Enable zlib compression support"
               OPTIONS BUNDLED SYSTEM OFF
               DEFAULT VALUE BUNDLED)
+mongo_setting(USE_BUNDLED_UTF8PROC "Enable building with utf8proc. Needed for SCRAM-SHA-256 authentication with non-ASCII passwords"
+              OPTIONS TRUE FALSE
+              DEFAULT VALUE TRUE
+              ADVANCED)
```

Comment:
To test the system install, I ran `sudo apt-get install libutf8proc-dev` (`brew install utf8proc` on Mac), manually set `USE_BUNDLED_UTF8PROC` to `FALSE`, and included the library in a random file to make sure that CMake was able to configure and build correctly.